ivy/__init__.py

```diff
@@ -221,13 +221,22 @@ def __repr__(self):
             f"ivy.Shape({shape_repr})" if self._shape is not None else "ivy.Shape(None)"
         )
 
-    def __add__(self, other):
+    def __add__(self, other, reverse=False):
         try:
-            self._shape = self._shape + other
+            if reverse:
+                self._shape = other + self._shape
+            else:
+                self._shape = self._shape + other
         except TypeError:
-            self._shape = self._shape + list(other)
+            if reverse:
+                self._shape = list(other) + self._shape
+            else:
+                self._shape = self._shape + list(other)
         return self
 
+    def __radd__(self, other):
+        return self.__add__(other, reverse=True)
```

Comment:
Let's just add the same logic in the reverse way, rather than adding a new argument to `__add__`. Would just make the code more readable 🙂

Reply:
sounds good